BUG: Entrypoints changed from dict to special object in py3.12

## Current Behavior
Tests have been failing on py3.12 due to an entrypoint api change
<details><summary>Example test failure</summary>
<p>

```
entry_points = (EntryPoint(name='pcdsdevices', value='pcdsdevices.happi.containers', group='happi.containers'), EntryPoint(name='pcds... group='console_scripts'), EntryPoint(name='qta-browser', value='qtawesome.icon_browser:run', group='console_scripts'))

    def test_typhos_entrypoint(entry_points):
>       assert "typhos.ui" in entry_points, "No typhos entrypoints"

pcdsdevices/tests/test_entrypoints.py:18: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = EntryPoint(name='pcdsdevices', value='pcdsdevices.happi.containers', group='happi.containers')
other = 'typhos.ui'

    def __eq__(self, other):
>       return self._key() == other._key()
E       AttributeError: 'str' object has no attribute '_key'

/opt/hostedtoolcache/Python/3.12.3/x64/lib/python3.12/importlib/metadata/__init__.py:258: AttributeError
```

</p>
</details> 

## Expected Behavior
tests should work

## Context / environment
py3.12 CI/CD

## Steps to Reproduce (for bugs)
1. Install pcdsdevices in a python 3.12 environment
2. run tests

## Suggested Solution
Make entrypoint agnostic to different versions

--- a/docs/source/upcoming_release_notes/1230-tst_entrypoint_312.rst
+++ b/docs/source/upcoming_release_notes/1230-tst_entrypoint_312.rst
@@ -1,0 +1,30 @@
+1230 tst_entrypoint_312
+#######################
+
+API Breaks
+----------
+- N/A
+
+Features
+--------
+- N/A
+
+Device Updates
+--------------
+- N/A
+
+New Devices
+-----------
+- N/A
+
+Bugfixes
+--------
+- N/A
+
+Maintenance
+-----------
+- Modifies entrypoint tests to be forward-compatible with py3.12 entrypoint API
+
+Contributors
+------------
+- tangkong

--- a/pcdsdevices/tests/test_entrypoints.py
+++ b/pcdsdevices/tests/test_entrypoints.py
@@ -2,6 +2,7 @@ from __future__ import annotations
 
 import importlib.metadata
 import pathlib
+import sys
 
 import pytest
 import typhos
@@ -11,6 +12,10 @@ from .. import ui
 
 @pytest.fixture(scope="session")
 def entry_points() -> dict[str, importlib.metadata.EntryPoints]:
+    if sys.version_info[1] >= 12:
+        # re-organize entrypoints to match old
+        eps = importlib.metadata.entry_points()
+        return {group_name: eps.select(group=group_name) for group_name in eps.groups}
     return importlib.metadata.entry_points()
 
 

